app/code/Magento/Shipping/Model/Shipping.php

```diff
@@ -234,11 +234,7 @@ public function collectRates(\Magento\Quote\Model\Quote\Address\RateRequest $req
 
         $limitCarrier = $request->getLimitCarrier();
         if (!$limitCarrier) {
-            $carriers = $this->_scopeConfig->getValue(
-                'carriers',
-                \Magento\Store\Model\ScopeInterface::SCOPE_STORE,
-                $storeId
-            );
+            $carriers = $this->_shippingConfig->getCarriersConfig($storeId);
 
             foreach ($carriers as $carrierCode => $carrierConfig) {
                 $this->collectCarrierRates($carrierCode, $request);
```

Comment:
just curious why not use method getAllCarriers

Reply:
seem i see some regression code here with change public to private method

Reply:
@mrtuvn you mean private to public, that's not a regression

Reply:
We don't use the getAllCarriers method because we don't need the carrier model to be instantiated